src/containers/CouponDetails/CouponDetails.test.jsx

```diff
@@ -120,6 +120,7 @@ describe('CouponDetailsWrapper', () => {
     });
 
     it('with error', () => {
+      // TODO: set selectedToggle: unredeemed state on component mount
```

Comment:
@adamstankiewicz @muhammad-ammar Any thoughts how to set component only state here?

Reply:
@mushtaqak You could try something like this:
```jsx
import CouponDetailsComponent from '../../components/CouponDetails';
// ...

it('with error', () => {
  const tree = renderer.create((
    <CouponDetailsWrapper
      couponData={{
        ...initialCouponData,
        errors: [{ code: 'test-code', user_email: 'test@example.com' }],
      }}
      isExpanded
    />
  ));
  tree.root.findByType(CouponDetailsComponent).instance.setState({ selectedToggle: 'unredeemed' });
  expect(tree.toJSON()).toMatchSnapshot();
});
```

Reply:
This works 👍 

even though I had tried this approach before but this didn't work ... there is only one exception I was finding the container `CouponDetails` whereas it had to be the underlying CouponDetails component `import CouponDetailsComponent from '../../components/CouponDetails';` ... ughhh miner mistakes  🤕

Reply:
Glad it works! Yeah, if you just look for `CouponDetails`, it's actually finding the container, not the component itself. The rest of the PR looks good, though; I'll give this a 👍once the test is updated :)